controllers/openstackbaremetalset_controller.go

```diff
@@ -223,26 +252,85 @@ func (r *OpenStackBaremetalSetReconciler) Reconcile(ctx context.Context, req ctr
 		r.Log.Info(fmt.Sprintf("OpenStackIPSet for %s successfully reconciled - operation: %s", instance.Name, string(op)))
 	}
 
-	if len(ipset.Status.HostIPs) < instance.Spec.Replicas {
-		r.Log.Info(fmt.Sprintf("OpenStackIPSet has not yet reached the required replicas %d", instance.Spec.Replicas))
-		return ctrl.Result{RequeueAfter: time.Second * 10}, nil
+	if len(ipset.Status.HostIPs) < instance.Spec.Count {
+		msg := fmt.Sprintf("OpenStackIPSet has not yet reached the required count %d", instance.Spec.Count)
+		instance.Status.ProvisioningStatus.State = ospdirectorv1beta1.BaremetalSetWaiting
+		instance.Status.ProvisioningStatus.Reason = msg
+		r.Log.Info(msg)
+
+		err2 := r.setStatus(instance)
+		return ctrl.Result{RequeueAfter: time.Second * 10}, err2
 	}
 
 	// Provision / deprovision requested replicas
-	if err := r.ensureBaremetalHosts(instance, provisionServer, sshSecret, ipset); err != nil {
+	if err := r.ensureBaremetalHosts(instance, provisionServer, sshSecret, passwordSecret, ipset); err != nil {
+		// As we know, this transient "object has been modified, please try again" error occurs from time
+		// to time in the reconcile loop.  Let's avoid counting that as an error for the purposes of
+		// status display
+		// FIXME?: Is there an "errors.IsWhatever()" func we can call for this type of error?
+		if !strings.Contains(err.Error(), "please apply your changes to the latest version and try again") {
```

Comment:
if we hit this and don't want to reconcile with an error, we might want to reconcile without logging an error and do a manual retry like:
   return ctrl.Result{RequeueAfter: time.Second * 1}, nil

Instead of hard code this string here, we can use the OptimisticLockErrorMsg const from
https://pkg.go.dev/k8s.io/apiserver/pkg/registry/generic/registry#pkg-constants

Reply:
Ah, thank you!  That seems to fit what I was looking for in the `FIXME`.